### PR TITLE
Docs update for nlp refactoring

### PR DESCRIPTION
**Pull request by VahidooX**
Updated documentations for 
+ Intent detection and slot filling
+ Transformer language modelling
+ Losses

### CODE REVIEW

File: README.rst [hunk 1]
```diff
@@ -39,7 +39,16 @@ NeMo consists of:
...
+* 1) **NLP Collection**
+* Intent Detection and Slot Filling
+* Text Classification
+* State Tracking for Task-oriented Dialogue Systems
+* Language Modelling
+* Machine Translation
+* Question Answering
+* Name Entity Recognition
+* Punctuation and Capitalization
+* GLUE Benchmark
```
Comment (yzhang123) [inline README.rst]: what about asr_postprocessing?

File: docs/sources/source/nlp/joint_intent_slot_filling.rst [hunk 2]
```diff
@@ -128,17 +126,14 @@ Next, we define all Neural Modules participating in our joint intent slot fillin
...
-                local_rank=local_rank,
                 ignore_extra_tokens=args.ignore_extra_tokens,
-                ignore_start_end=args.ignore_start_end
-                )
+                ignore_start_end=args.ignore_start_end,
+            )
 
-            ids, type_ids, input_mask, loss_mask, \
-                subtokens_mask, intents, slots = data_layer()
+            (ids, type_ids, input_mask, loss_mask, subtokens_mask, intents, slots) = data_layer()
```
Comment (yzhang123) [inline docs/sources/source/nlp/joint_intent_slot_filling.rst]: please use the Toms port refactoring, which is input_data = data_layer(), input_data.ids, input_data.type_ids etc

File: docs/sources/source/nlp/joint_intent_slot_filling.rst [hunk 3]
```diff
@@ -128,17 +126,14 @@ Next, we define all Neural Modules participating in our joint intent slot fillin
...
             data_size = len(data_layer)
 
-            logging.info(f'The length of data layer is {data_size}')
+            print(f'The length of data layer is {data_size}')
```
Comment (yzhang123) [inline docs/sources/source/nlp/joint_intent_slot_filling.rst]: please use logging

File: docs/sources/source/nlp/joint_intent_slot_filling.rst [hunk 4]
```diff
@@ -148,85 +143,73 @@ Next, we define all Neural Modules participating in our joint intent slot fillin
             steps_per_epoch = math.ceil(data_size / (batch_size * num_gpus))
             logging.info(f"Steps_per_epoch = {steps_per_epoch}")
 
-            hidden_states = pretrained_bert_model(input_ids=ids,
-                                                  token_type_ids=type_ids,
-                                                  attention_mask=input_mask)
+            hidden_states = pretrained_bert_model(input_ids=ids, token_type_ids=type_ids, attention_mask=input_mask)
```
Comment (yzhang123) [inline docs/sources/source/nlp/joint_intent_slot_filling.rst]: here use input_ids = data_layer().ids, etc

File: docs/sources/source/nlp/joint_intent_slot_filling.rst [hunk 5]
```diff
@@ -148,85 +143,73 @@ Next, we define all Neural Modules participating in our joint intent slot fillin
...
+        eval_tensors, _, _, data_layer = create_pipeline(
+            args.num_eval_samples, batch_size=args.batch_size, num_gpus=args.num_gpus, mode=args.eval_file_prefix,
+        )
 
     * Create relevant callbacks for saving checkpoints, printing training progresses and evaluating results.
 
     .. code-block:: python
 
-        train_callback = nemo.core.SimpleLossLoggerCallback(
+        train_callback = SimpleLossLoggerCallback(
```
Comment (yzhang123) [inline docs/sources/source/nlp/joint_intent_slot_filling.rst]: should we in the tutorial describe how to import these Classes? in that case nemo.core.SimpleLossLoggerCallback makes more sense

File: docs/sources/source/nlp/transformer_language_model.rst [hunk 6]
```diff
@@ -15,7 +15,7 @@ Download Corpus
 
 For demonstration purposes, we will be using the very small WikiText-2 dataset :cite:`nlp-lm-merity2016pointer`.
 
-To download the dataset, run the script ``examples/nlp/scripts/get_wt2.sh``. After downloading and unzipping, the folder should include 3 files that look like this:
+To download the dataset, run the script ``examples/nlp/scripts/get_wkt2.sh``. After downloading and unzipping, the folder should include 3 files that look like this:
```
Comment (yzhang123) [inline docs/sources/source/nlp/transformer_language_model.rst]: the script name is get_wt2.sh

File: examples/nlp/token_classification/token_classification.py [hunk 7]
```diff
@@ -48,7 +48,7 @@
 parser.add_argument("--ignore_start_end", action='store_false')
 parser.add_argument("--ignore_extra_tokens", action='store_false')
 parser.add_argument("--none_label", default='O', type=str)
-parser.add_argument("--do_not_shuffle_data", action='store_false')
+parser.add_argument("--shuffle_data", action='store_true')
```
Comment (yzhang123) [inline examples/nlp/token_classification/token_classification.py]: update this to recent PR

File: docs/sources/source/nlp/joint_intent_slot_filling.rst (hunk 2 above)
Comment (VahidooX) [reply docs/sources/source/nlp/joint_intent_slot_filling.rst]: Done.

File: docs/sources/source/nlp/joint_intent_slot_filling.rst (hunk 3 above)
Comment (VahidooX) [reply docs/sources/source/nlp/joint_intent_slot_filling.rst]: Done.

File: docs/sources/source/nlp/joint_intent_slot_filling.rst (hunk 4 above)
Comment (VahidooX) [reply docs/sources/source/nlp/joint_intent_slot_filling.rst]: Done.

File: docs/sources/source/nlp/joint_intent_slot_filling.rst (hunk 5 above)
Comment (VahidooX) [reply docs/sources/source/nlp/joint_intent_slot_filling.rst]: In tutorial we actually just explain what happens in the example. So they should check the example to understand whats happening otherwise no way. How to do that? Add the imports to the tutorial?

File: docs/sources/source/nlp/transformer_language_model.rst (hunk 6 above)
Comment (VahidooX) [reply docs/sources/source/nlp/transformer_language_model.rst]: It is get_wtk2.sh in my version. Checked the master too.

File: examples/nlp/token_classification/token_classification.py (hunk 7 above)
Comment (VahidooX) [reply examples/nlp/token_classification/token_classification.py]: Isn't it the right way? Which PR?

File: docs/sources/source/nlp/joint_intent_slot_filling.rst (hunk 5 above)
Comment (yzhang123) [reply docs/sources/source/nlp/joint_intent_slot_filling.rst]: i have added the imports for other tutorials

File: docs/sources/source/nlp/transformer_language_model.rst (hunk 6 above)
Comment (yzhang123) [reply docs/sources/source/nlp/transformer_language_model.rst]: ok, probably this was renamed sometime.

File: examples/nlp/token_classification/token_classification.py (hunk 7 above)
Comment (yzhang123) [reply examples/nlp/token_classification/token_classification.py]: why do you want to change this line? we had this discussion here already in https://github.com/NVIDIA/NeMo/pull/401

File: docs/sources/source/nlp/joint_intent_slot_filling.rst (hunk 5 above)
Comment (VahidooX) [reply docs/sources/source/nlp/joint_intent_slot_filling.rst]: Done.

File: examples/nlp/token_classification/token_classification.py (hunk 7 above)
Comment (VahidooX) [reply examples/nlp/token_classification/token_classification.py]: If you set the default to store_false, then when use --shuffle_data it would disable the shuffling. The default for such args is False and I am not sure how to have True as default. Do you suggest to change the name to do_not_shuffle_data?

Comment (yzhang123) [reply examples/nlp/token_classification/token_classification.py]: yes, basically the outcome of #401 is to use --do_not_shuffle_data", action='store_false'. This will by default be true, and only false if you explicitly pass --do_not_shuffle_data

Comment (VahidooX) [reply examples/nlp/token_classification/token_classification.py]: Changed it to :
parser.add_argument("--no_shuffle_data", action='store_false', dest="shuffle_data")

File: README.rst (hunk 1 above)
Comment (VahidooX) [reply README.rst]: Added.
